espnet/nets/pytorch_backend/transformer/attention.py

```diff
@@ -204,3 +204,97 @@ def forward(self, query, key, value, pos_emb, mask):
         )  # (batch, head, time1, time2)
 
         return self.forward_attention(v, scores, mask)
+
+
+class SlidingWindow(nn.Module):
+    """Sliding window for sequence.
+
+    Paper: https://arxiv.org/abs/1909.13037
+    Fig 2
+
+    Args:
+        n_feat (int): The number of features.
+        left_context (int): Sliding windows left length
+        right_context (int): Sliding windows right length
+
+    """
+
+    def __init__(self, n_feat, left_context=5, right_context=2):
+        """Construct a SlidingWindow object."""
+        super(SlidingWindow, self).__init__()
+        self.n_feat = n_feat
+        self.left_context = left_context
```

Comment:
Btw, you should maybe assert left/right context have correct values? You only check value is != 0 during block creation

Reply:
what means of "correct value", from my opinion, all >0 int value is "correct"

Reply:
Negative values and kernel size bigger than input size are allowed/handled?

Reply:
I don't understand which kernel you specified?

Reply:
Sorry, wanted to say `chunk_size` (left-window, right-window)!

Reply:
since chunk_size is the sum of left and right window, I just add a check with these two.